src/compiler.ts

```diff
@@ -8439,17 +8446,23 @@ export class Compiler extends DiagnosticEmitter {
         }
         this.currentType = localType;
 
-        if (target.parent != flow.parentFunction) {
-          // TODO: closures
-          this.error(
-            DiagnosticCode.Not_implemented_0,
-            expression.range,
-            "Closures"
-          );
-          return module.unreachable();
-        }
+        //if (target.parent != flow.parentFunction) {
+          //// TODO: closures
+          //this.error(
+            //DiagnosticCode.Not_implemented_0,
+            //expression.range,
+            //"Closures"
+          //);
+          //return module.unreachable();
+        //}
```

Comment:
This should be unreachable now, since in this case we would return a `CLOSED_LOCAL` instead

Reply:
This would be a better place than inside of `lookup` for obtaining closed locals, for example by means of `Function#ensureClosedLocal` as mentioned below. I am also not sure if the `lookup` above is able to return scoped locals, or if it just returns top-level locals of a function, so that might have been tackled as well. Then also leads back to the challenge that scoped locals are shared between different `let`s. Scenarios: Local has been used for a different `let` before: Unable to figure out which local accesses to replace when replacing with env accesses. Local will be used by a different `let` after: Not the same as in the env.

Reply:
Hm- could you give a code example of this scoped local issue?

Reply:
A simple example for the reuse problem is

```ts
{
  let a = 1;
}
{
  let b = 2;
}
```

where both `a` and `b` are of the same type. At the end of the first block, `a` is free'd, and upon `let b` the underlying local incl. its index is reused.

The lookup problem is that

```ts
var a;
someFunc() {
  a
}
```

works, but

```ts
{
  let a;
  someFunc() {
    a
  }
}
```

may not. Didn't check in detail, but iirc the latter one doesn't even trigger `Not implemented: Closures` currently.